src/main/java/uk/gov/hmcts/darts/arm/mapper/impl/AnnotationArchiveRecordMapperImpl.java

```diff
@@ -65,43 +107,201 @@ private void loadAnnotationProperties() throws IOException {
     private AnnotationArchiveRecord createAnnotationArchiveRecord(AnnotationCreateArchiveRecordOperation annotationCreateArchiveRecordOperation,
                                                                   UploadNewFileRecord uploadNewFileRecord) {
         return AnnotationArchiveRecord.builder()
-                .annotationCreateArchiveRecordOperation(annotationCreateArchiveRecordOperation)
-                .uploadNewFileRecord(uploadNewFileRecord)
-                .build();
+            .annotationCreateArchiveRecordOperation(annotationCreateArchiveRecordOperation)
+            .uploadNewFileRecord(uploadNewFileRecord)
+            .build();
     }
 
-    private UploadNewFileRecord createUploadNewFileRecord(AnnotationDocumentEntity annotationDocument, Integer relationId) {
+    private AnnotationCreateArchiveRecordOperation createArchiveRecordOperation(ExternalObjectDirectoryEntity externalObjectDirectory) {
+        return AnnotationCreateArchiveRecordOperation.builder()
+            .relationId(String.valueOf(externalObjectDirectory.getId()))
+            .recordMetadata(createArchiveRecordMetadata(externalObjectDirectory))
+            .build();
+    }
+
+    private RecordMetadata createArchiveRecordMetadata(ExternalObjectDirectoryEntity externalObjectDirectory) {
+        AnnotationDocumentEntity annotationDocument = externalObjectDirectory.getAnnotationDocumentEntity();
+        RecordMetadata metadata = RecordMetadata.builder()
+            .publisher(armDataManagementConfiguration.getPublisher())
+            .recordClass(armDataManagementConfiguration.getAnnotationRecordClass())
+            .recordDate(currentTimeHelper.currentOffsetDateTime().format(dateTimeFormatter))
+            .eventDate(formatDateTime(annotationDocument.getUploadedDateTime()))
+            .region(armDataManagementConfiguration.getRegion())
+            .title(annotationDocument.getFileName())
+            .clientId(String.valueOf(externalObjectDirectory.getId()))
+            .build();
+
+        String courthouse = getCourthouse(annotationDocument);
+        String courtroom = getCourtroom(annotationDocument);
+        if (nonNull(courthouse) && nonNull(courtroom)) {
+            metadata.setContributor(courthouse + " & " + courtroom);
+        }
+
+        if (annotationRecordProperties.containsKey(BF_001_KEY)) {
```

Comment:
possible to replace all these with a hashmap of methods? e.g
```
MetaData metaData = new MetaData();
Map<String, Consumer<String>> commands = new HashMap<>();
commands.put(BF_001_KEY, metaData::setBf001);
commands.put(BF_002_KEY, metaData::setBf002);
```

Reply:
The hashmap is a map of the properties which differ between live, nle for each media, case document, annotation document and transcription document. The values may or may not be relevant as well as be different in live to nle. We don't want to be trying to set attributes that don't exist in the property file.